src/libraries/System.IO.FileSystem/tests/FileStream/ctor_options.Unix.cs

```diff
@@ -15,20 +15,18 @@ private static long GetAllocatedSize(FileStream fileStream)
             using var px = Process.Start(new ProcessStartInfo
             {
                 FileName = "stat",
-                ArgumentList = { isOSX ? "-f"    : "-c",
-                                 isOSX ? "%b %k" : "%b %B",
+                ArgumentList = { PlatformDetection.IsBsdLike ? "-f"    : "-c",
+                                 PlatformDetection.IsBsdLike ? "%z" : "%b %B",
                                  fileStream.Name },
                 RedirectStandardOutput = true
             });
             string stdout = px.StandardOutput.ReadToEnd();
 
             string[] parts = stdout.Split(' ');
-            return long.Parse(parts[0]) * long.Parse(parts[1]);
+            return PlatformDetection.IsBsdLike ? long.Parse(parts[0]) : long.Parse(parts[0]) * long.Parse(parts[1]);
         }
 
-        private static bool SupportsPreallocation =>
-            RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ||
-            RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
+        private static bool SupportsPreallocation => OperatingSystem.IsLinux() || PlatformDetection.IsBsdLike;
```

Comment:
I am not sure whether this is true. `fallocate` is Linux-specific sys-call and afaik `F_PREALLOCATE` is not defined on FreeBSD?

https://github.com/dotnet/runtime/blob/a24364a09d9aea98b545f16689a53bafc6b18c14/src/native/libs/System.Native/pal_io.c#L1052-L1062

Reply:
Correct; FreeBSD does not have an `F_PREALLOCATE` only an `fallocate()` and `posix_fallocate()` if `HAVE_POSIX_FALLOCATE`. I don't believe the POSIX is guaranteed present either.

Reply:
rolled back the change `SupportsPreallocation` will be false on FreeBSD.